pkg/virt-api/webhooks/validating-webhook/admitters/vmi-create-admitter.go

```diff
@@ -220,8 +220,12 @@ func validateNetworksMatchInterfaces(field *k8sfield.Path, spec *v1.VirtualMachi
 	portForwardMap := make(map[string]struct{})
 
 	vifMQ = spec.Domain.Devices.NetworkInterfaceMultiQueue
-	isVirtioNicRequested = false
 
+	if len(spec.Networks) == 0 && len(spec.Domain.Devices.Interfaces) == 0 {
+		return networkInterfaceMap, vifMQ, true, causes, done
```

Comment:
I really do not like this early exit, nor the fact the we ask the same question twice.
The problem is with this ugly variable that is used to piggy-back on the for-loop. This whole function has a smell, telling us it does too many things.

This addition just exposes the problem that is already here and cleaning it up will be beneficial.

Reply:
Extracted the relevant code to a function that does only this thing
removed the `len(spec.Networks) == 0` which was there just because we wanted to do the same condition as autoAttach does (it was redundant but symetric), but now since we don't depend on autoAttach it is even more redundant